workflows/src/main/kotlin/com/template/flows/DenyAppointmentRequest.kt

```diff
@@ -36,37 +38,23 @@ class DenyAppointmentRequest(private val alice: Party,
         //Compose the State that carries the appointment information
 
 
-        // Step 3. Create a new TransactionBuilder object.
+        // Step 3. Create a new TransactionBuilder object. The request is going to be consumed, but the availableDate will
+        //         continue to be available
         val builder = TransactionBuilder(notary)
-                .addCommand(AppointmentContract.Commands.Create(), listOf(doctor.owningKey, alice.owningKey))
+                        .addInputState(request)
+                        .addCommand(AppointmentRequestContract.Commands.Deny(), listOf(doctor.owningKey))
 
 
         // Step 4. Verify and sign it with our KeyPair.
         builder.verify(serviceHub)
         val ptx = serviceHub.signInitialTransaction(builder)
 
-        val states = serviceHub.vaultService.queryBy(Appointment::class.java)
-        //get state from states using id
-        var output: Appointment? = null
-        for(state: StateAndRef<Appointment> in states.states){
-            if(date == state.state.data.date){
-                output=state.state.data
-            }
-        }
-
-        if(output==null){
-            throw StatesNotAvailableException("State not Found")
-        }
-
         // Step 6. Collect the other party's signature using the SignTransactionFlow.
-        val otherParties: MutableList<Party> = output.participants.stream().map { el: AbstractParty? -> el as Party? }.collect(Collectors.toList())
-        otherParties.remove(ourIdentity)
+        val otherParties: MutableList<Party> = listOf(alice).stream().map { el: AbstractParty? -> el as Party? }.collect(Collectors.toList())
```

Comment:
If I'm not missing something you should be able to change that to "mutableListOf(alice)"